spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/DefaultTokenServices.java

```diff
@@ -291,7 +291,7 @@ private OAuth2RefreshToken createRefreshToken(OAuth2Authentication authenticatio
 		}
 		int validitySeconds = getRefreshTokenValiditySeconds(authentication.getOAuth2Request());
 		String tokenValue = new String(Base64.encodeBase64(
-				DEFAULT_TOKEN_GENERATOR.generateKey()));
+				DEFAULT_TOKEN_GENERATOR.generateKey(), true, true));
```

Comment:
Can you also please add a test.

Reply:
Hi @jgrandja, of course I can write a test. Can you give me a tip on what the test should cover?

Reply:
After looking into this, I realized that it's hard to provide a test since there is setter for overriding `DEFAULT_TOKEN_GENERATOR`. And I'd rather not expose a setter at this point. I think it's safe to not provide a test for this instance.

Also, there were a few failing tests as a result of `isChunked=true` so I updated to call `encodeBase64URLSafe()` instead.